### ISSUE TITLE
Table: Filtering items in table crashes app on Safari

### ISSUE THREAD

**Issue by AndrianStoikov** [SAP/ui5-webcomponents#6570]
**Describe the bug**
When the items of the `Table` component change, the whole application crashes on Safari. 

**Isolated Example**
[sandbox](https://codesandbox.io/s/optimistic-villani-1gi8yo?file=/src/App.js)

**To Reproduce**
1. Go to the sandbox
2. Click on the input
3. Type something
4. The page crashes

**Expected behavior**
The `Table` component should be able to handle element filtering

**Screenshots**

**UI5 Web Components for React Information**
`@ui5/webcomponents` version: latest (1.10.3)
`@ui5/webcomponents-react` version: 1.8.1
Operating System: Mac OS Ventura 13.2.1
Browser: Safari Version 16.3 (18614.4.6.1.6)

**Additional context**
When the keys in React are not unique the page works correctly. Unfortunately, once the keys are switched to unique(as it should be) the whole application crashes. The bug happens both on iPhone and Mac.

**Comment by AndrianStoikov** [comment]
Hi @Lukas742,

Thanks for pointing out the optimization. Also to add to the issue. We also couldn't debug what happens and it was very hard to understand that the table causes the page crash. It's very interesting because the logic is quite simple and it works perfectly on Chrome.

**Comment by Lukas742** [comment]
@AndrianStoikov a colleague just pointed out to me that it's possible to find the crash report in the `Console` app on MacOS, but the error is very cryptic:

Excerpt of the error:
```
Crashed Thread:        0  Dispatch queue: com.apple.main-thread

Exception Type:        EXC_BAD_ACCESS (SIGSEGV)
Exception Codes:       KERN_INVALID_ADDRESS at 0x0000000000000018
Exception Codes:       0x0000000000000001, 0x0000000000000018

Termination Reason:    Namespace SIGNAL, Code 11 Segmentation fault: 11
Terminating Process:   exc handler [11671]

VM Region Info: 0x18 is not in any region.  Bytes before following region: 4311613416
      REGION TYPE                    START - END         [ VSIZE] PRT/MAX SHRMOD  REGION DETAIL
      UNUSED SPACE AT START
--->  
      __TEXT                      100fe0000-100fe4000    [   16K] r-x/r-x SM=COW  ...it.WebContent
```

**Comment by AndrianStoikov** [comment]
Hi @ilhan007,
Can someone from the team assist us with this bug? Currently, all our safari users are seriously affected by this bug.

**Comment by niyap** [comment]
Hello Colleagues,

Please, test whether the issue is reproducible from your side and do a pre-analysis before forwarding the issue to our component as there is nothing described and following the steps to reproduce in the description of the issue, I am not able to reproduce the issue. In Safari, the isolated sample works in the same way as in Chrome from my side.
Safari version: Version 16.3 (18614.4.6.1.6)
MacOS version: Ventura 13.2.1

Kind Regards,
Niya

**Comment by AndrianStoikov** [comment]
Hello @niyap,

The example that I provided is tested and it definitely does not work on Safari. I just tested in on my phone and when I type `K` in the search box it does break. Multiple colleagues tested it on their Macbooks and it breaks on them. Can you please try again?

**Comment by NHristov-sap** [comment]
Hi @niyap,

I have checked it and it is reproducible. Steps to reproduce:

1. Go to https://codesandbox.io/s/frosty-snowflake-07yfln?file=/src/App.js (Safari browser)
2. Focus the input
3. ...
3A. Type "S" inside -> works ... (then delete "S")
3B. Type "C" inside (on empty Input) -> the page is reloaded/crashed

I have checked it with a colleague, he reproduced it too

Best Regards,
Nikolay Hristov

**Comment by hristop** [comment]
Hi @Lukas742 , @AndrianStoikov ,

What is missing for me is a code snippet with only web components and no react. Can you please provide such where the issue can be reproduced only with web-components and no react at all? 

Best Regards,
Hristo

**Comment by AndrianStoikov** [comment]
Hello @hristop,

We have always provided examples with react because that is the technology we use. I don't know how to create the sandbox only with html and css and it's not even close to our productive setup. @Lukas742 has already provided a sandbox without the react wrapper so the problem is clearly not on the react side. I don't understand what is the problem at the moment ...

**Comment by Lukas742** [comment]
I wasn't able to reproduce the issue with plain js, but since it is still crashing the page in React, could you please check if "Medium" is really the right priority for this issue? Thanks.

https://codesandbox.io/s/ui5-webcomponents-forked-ftsrul?file=/src/index.js

**Comment by Lukas742** [comment]
I think without the `key` it won't break as it doesn't force the component to rerender. Every time the `key` of a given item in the list changes, React knows that something has changed and that it has to update the component. So the issue here has probably something to do with the update cycle of React in combination with how the `Table` handles updates. 

I just tested rendering only the `TableRow`s (without the `Table`) and then it works fine:
https://codesandbox.io/s/keen-paper-eku3n2?file=/src/App.js

**Comment by ilhan007** [comment]
Yep, it seems so. We have to dig deeper.

I also checked that the List is working in the same setup (we have cell in a row in a table, while here just item in a list)
```tsx
<List>
        {toRender.map((el) => (
          <StandardListItem key={el.id}>
            {el.id} {el.name}
          </StandardListItem>
        ))}
</List>
 ```

**Comment by Lukas742** [comment]
I tried two more things:
My first try was inlining the filter function, so that we can omit the `useEffect`. Unfortunately, the result is still the same:
https://codesandbox.io/s/great-hill-2et7te

The same applies when adding the event listener without React and removing the `value` prop from the input:
https://codesandbox.io/s/fragrant-dust-jv65fh?file=/src/App.js

**Comment by ilhan007** [comment]
Hello colleagues @SAP/ui5-webcomponents-topic-rl
I continued working on the issue and have some findings

I created very small, almost blank project, with `React + pure ui5 web components` (ui5-table, ui5-table-column, ui5-table-row and ui5-table-cell). I hope you can access this [link](https://sap-my.sharepoint.com/:u:/p/ilhan_myumyun/ESgyG-k56hBNtQ4MIeGAC94BYjkK37QW4iWBLKAUXHhSdA?e=PLie3T) to the project and download it if necessary.

As in all provided samples so far, we filter a table upon typing:
<img width="664" alt="Screenshot 2023-03-09 at 14 26 19" src="https://user-images.githubusercontent.com/15702139/224022676-0f32dfe9-ea78-4f53-9b09-3900116c7f80.png">
And, in all cases the filtering works as long as the first item is not filtered out before the second, when that happens the `Safari` browser reloads as described in the issue (and, `it breaks when we type "b"`, because this is the only way to get the first item filtered before the second one). In this scenario React seems to perform much more task of removing, adding DOM elements to the slots.

**At first,** we stopped injecting any styles to the TableCell|TableRow| shadow Dom as we thought for Safari we use `<style> ` tags (not addoptedStylesheets as in Chrome, because it's still not supported on Safari) and that might causing this weird behaviour. And the issue was gone. The table looked ugly, but the filtering worked properly.
**Then,** we started returning the styles, and removing small parts until we tracked down to `"overflow: hidden"` of the `TableCell` styles. Without "overflow: hidden", everything works properly on Safari, with "overflow: hidden" the issue is reproducible.

--------
**How I tested it**
In my [simple app](https://sap-my.sharepoint.com/:u:/p/ilhan_myumyun/ESgyG-k56hBNtQ4MIeGAC94BYjkK37QW4iWBLKAUXHhSdA?e=PLie3T) I linked the ui5 web components project (the "main" package) and started playing with the styles of the TableCell. They can be found in `ui5-webcomponents/packages/main/dist/TableCell.js`. As I mentioned, started removing styles until I tracked the root cause down to the "overflow: hidden".

I don't have explanation why this happens. It should be combination of the React + Safari factor.
Nevertheless, by design UI5 Web Components (the example is with pure web components) should work in React and I think we need to work on a solution. It seems we need to find workaround for that style on Safari. I will be glad to help you whatever needed in case you have questions. I hope this helps you continue with the issue.

--------

- **This is how it breaks**
https://user-images.githubusercontent.com/15702139/224025707-df16ca58-75f3-447d-b516-f34a80d3c173.mov

- **This is how it works**  when removing the `overflow: hidden` from `ui5-webcomponents/packages/main/dist/TableCell.js`

https://user-images.githubusercontent.com/15702139/224025849-c138fb52-d6dc-42ce-8ba3-dcc53584d8e0.mov


BR,
ilhan

**Comment by AndrianStoikov** [comment]
Hello Team,

Can we have an update on this issue? Our product is pretty much unusable at the moment in Safari. Can we please increase the priority of this issue? Also, another option is opening an issue in Safari, if it is possible.

Best,
Andrian

**Comment by Neeeko** [comment]
Hi, any update on this issue? We are also encountering a similar issue in a SuccessFactors project. Thanks.

**Comment by ilhan007** [comment]
Hello @SAP/ui5-webcomponents-topic-rl colleagues  - changed the priority to High. The Table is not really usable on Safari. You can refer to the findings so far, especially [one of the last comments]( https://github.com/SAP/ui5-webcomponents/issues/6570#issuecomment-1461991374) to get you going.

**Comment by ndeshev** [comment]
I tried to find information for the issue on the web and couldn't, but there are similar issues that have been reported for Safari crashing the same way with other styles/combination of styles in the past.

I've test the removing of the style on the provided sample app and it seems to work fine, also it doesn't seem to break the visual appearance of the component.

I created a pull-request that doesn't add this style in case the detected browser is Safari

**Comment by ilhan007** [comment]
Hello @AndrianStoikov @Lukas742 the issue has been fixed and released with the latest 1.13.0-rc.1 https://github.com/SAP/ui5-webcomponents/releases/tag/v1.13.0-rc.1

@Lukas742 probably we need to align how to make it available via the ui5-webcompoents-react package
